### PR TITLE
fix(ci): skip watchedFiles test, build in-container

### PR DESCRIPTION
**Pull request by justinmk3**
CI tests are failing ~~since 5e40506dd627.~~ That correlates with...

1. SAM integ test failures started after release of SAM CLI 1.48: https://github.com/aws/aws-sam-cli/releases/tag/v1.48.0
2. `CloudFormationTemplateRegistry` test started failing after vscode 1.67 was released on 5/5

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by JadenSimon** [comment]
CI is just failing because Python is missing `distutils` in the environment.
> ModuleNotFoundError: No module named 'distutils.util'

Try `sudo apt-get install python3-distutils` in the buildspec? Not sure if it's still available though.

**Comment by justinmk3** [comment]
distutils is deprecated, not sure why we need it at all

**Comment by justinmk3** [comment]
oh it's SAM CLI :(

**Comment by JadenSimon** [comment]
This is an interesting comment https://github.com/aws/aws-lambda-builders/blob/8a30554603bf96a4f9bea92866134a1015d197c2/aws_lambda_builders/workflows/python_pip/packager.py#L149-L154

Our CI uses Python 3.9 for each version. Not sure if it matters though. I think `pip` uses a specific `distutils` version for each Python version.

**Comment by justinmk3** [comment]
I wonder if doing a container build could workaround this

**Comment by JadenSimon** [comment]
> Error: Timed out waiting for registry to populate. Actual item count: 1. Expected item count: 2.

Ah right this integration test is broke too. The file watcher might be flawed but I can't remember.

**Comment by justinmk3** [comment]
> Ah right this integration test is broke too. The file watcher might be flawed but I can't remember.

[SAM CLI 1.48](https://github.com/aws/aws-sam-cli/releases/tag/v1.48.0) was released just a few minutes before this CI issue started :)

**Comment by justinmk3** [comment]
```
  1) CloudFormation Template Registry
       adds dynamically-added template files with yaml and yml extensions at various nesting levels:
     Error: Timed out waiting for registry to populate. Actual item count: 1. Expected item count: 2.
```

On 5/4 (5e40506dd627258dbd85d14fbdbd04050b5e9db1) this test was _only_ failing on Insiders.

vscode 1.67 was released on 5/5 (yesterday), now that test consistently fails on both Stable and Insiders.

**Comment by JadenSimon** [comment]
Maybe it's a subtle Linux-only bug? Test started failing ~28 days ago

**Comment by JadenSimon** [comment]
Still failing.

**Comment by justinmk3** [comment]
off topic, but noticed this in "minimum" CI:

```
globalSetup: before()
  CloudFormation Template Registry
    ✔ adds initial template files with yaml and yml extensions at various nesting levels (56ms)
2022-05-10 02:17:05 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a new file: /codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/dist/src/testFixtures/workspaceFolder/0/test.yaml
2022-05-10 02:17:05 [VERBOSE]: CloudFormationTemplateRegistry: manager detected a new file: /codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/dist/src/testFixtures/workspaceFolder/0/nested/test.yml
    - adds dynamically-added template files with yaml and yml extensions at various nesting levels
rejected promise not handled within 1 second: TypeError: Cannot read property 'registerContributor' of undefined
stack trace: TypeError: Cannot read property 'registerContributor' of undefined
    at /codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/src/shared/extensions/yaml.ts:46:21
    at Generator.next (<anonymous>)
    at fulfilled (/codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/dist/src/shared/extensions/yaml.js:5:808)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
Cannot read property 'registerContributor' of undefined: TypeError: Cannot read property 'registerContributor' of undefined
    at /codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/src/shared/extensions/yaml.ts:46:21
    at Generator.next (<anonymous>)
    at fulfilled (/codebuild/output/src071603141/src/github.com/aws/aws-toolkit-vscode/dist/src/shared/extensions/yaml.js:5:808)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)

File: src/shared/watchedFiles.ts [hunk 1]
```diff
@@ -51,14 +51,41 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
...
+     *
+     * @param glob Pattern to match against (across all opened workspace folders)
      */
     public async addWatchPattern(glob: vscode.GlobPattern): Promise<void> {
         if (this._isDisposed) {
             throw new Error(`${this.name}: manager has already been disposed!`)
         }
+        if (typeof glob === 'string' && !vscode.workspace.workspaceFolders?.[0]) {
+            getLogger().error(`${this.name}: no workspace, skipping addWatchPattern(${glob})`)
+            return
```
Comment (justinmk3) [inline src/shared/watchedFiles.ts:87]: wondering if this is the source of some of the bug reports we've seen

**Pull request merged**